src/blocks/gallery-masonry/styles/editor.scss

```diff
@@ -18,118 +17,119 @@
 		}
 	}
 
-	// Fixes issue where figures are set to display 100% if alignfull is set.
-	.coblocks-gallery--figure {
-		width: auto !important;
+	.gallery-media-placeholder-wrapper {
+		margin-top: 20px;
 
-		&.is-selected {
-			border-radius: $radius-block-ui + 2; // Additional because we modify the selected state.
+		.components-placeholder__fieldset {
+			align-items: center;
 
-			&::after {
-				border-radius: $radius-block-ui + 2;
-				box-shadow: inset 0 0 0 $border-width + 1 var(--wp-admin-theme-color, #007cba), inset 0 0 0 3px $white;
+			button {
+				margin-bottom: 0;
 			}
 		}
-
-		img {
-			vertical-align: bottom;
-		}
 	}
+}
 
-	// Fixes issue with multiple dropzones displaying when dropping in an image.
-	.components-drop-zone.is-active {
-
-		+ ul li:last-child .components-drop-zone {
-			display: none;
-		}
+figure.wp-block-coblocks-gallery-masonry {
+	&.alignwide,
+	&.alignfull {
+		margin: 0;
 	}
 
-	// Captions.
-	.block-editor-rich-text {
-		bottom: 0;
-		left: 0;
-		max-height: 100%;
-		overflow-y: auto;
-		position: absolute;
-		right: 0;
-		width: 100%;
-
-		.editor-rich-text__tinymce {
-			color: inherit;
-
-			a {
-				color: inherit;
+	> .masonry-grid {
+		column-fill: initial;
+		column-gap: 30px;
+		transition: all 0.5s ease-in-out;
+
+		figure {
+			display: block;
+			margin-bottom: 30px;
+			margin-top: 0;
+
+			img {
+				border: 0;
+				display: block;
+				font: inherit;
+				font-size: 100%;
+				height: auto;
+				margin: 0;
+				max-width: 100%;
+				padding: 0;
+				width: 100%;
 			}
 
-			&:not(.mce-content-body) {
-				opacity: 0.6;
+			figcaption {
+				background: linear-gradient(0deg, rgba($color: $black, $alpha: 0.7) 0, rgba($color: $black, $alpha: 0.3) 50%, transparent);
+				bottom: 0;
+				font-size: 13px;
+				margin-bottom: 0;
+				max-height: 100% !important;
+				opacity: 0.9;
+				overflow: auto;
+				padding: 30px 10px 10px !important;
+				position: absolute !important;
+				text-align: center;
+				width: 100%;
+
+				img {
+					display: inline;
+				}
 			}
 
-			&:focus a[data-mce-selected] {
-				opacity: 0.2;
+			// Fix needed for pre 5.9 WordPress
+			.components-resizable-box__container {
+				display: block;
+
+				.components-resizable-box__handle {
+					display: none;
+				}
 			}
 		}
-	}
 
-	.block-editor-rich-text figcaption:not([data-is-placeholder-visible="true"]) {
-		overflow: hidden;
-		position: relative !important;
-	}
-
-	.is-selected .block-editor-rich-text {
-		// IE calculates this incorrectly, so leave it to modern browsers.
-		@supports (position: sticky) {
-			bottom: 4px;
-			left: 4px;
-			margin-top: -4px;
-			right: 4px;
-			width: calc(100% - 8px);
+		/* Masonry on tablets */
+		@media only screen and (min-width: 768px) and (max-width: 1023px) {
+			& {
```

Comment:
I think (I might be wrong) you can remove lines 91 and 93. Same with 98 and 100

Reply:
Good call. This logic should exist only in the style.scss file and not here. This is duplicated code.

Reply:
Oh weird. I just tried removing these and the styles in the editor broke. I wonder if I didn't install the ones under styles.scss correctly. 🤔

Reply:
Oh, I got it now. I ended up having some duplicated code because we are targeting differing hierarchies of DOM selectors in the edit and the save markup. Because of this variation, the solution I went for was to include the same styles in both the editor.scss and style.scss file but using the applicable selector. I've updated these files to include a comment about the code duplication and why it exists so that we might improve it in the future.